### PR TITLE
Solution for small cases by rarias-p

### PR DESCRIPTION
**Pull request by TH0RlN**
First of all I calculate all the possible combinations of shelves (this is the least efficient way and the reason why the program does not work for bigger cases). Then I got the length of the books and of each combination of shelves. I discard all the combinations with a shorter length than the books. Finally I get the combination with the smallest number of shelves.

### CODE REVIEW

**Comment by aguimbao** [comment]
Please, revert your last 3 commits and make a new PR for challenge 05.

**Comment by clafoutis42** [comment]
You are deleting files from others... You have until 20.42 to make a clean PR, or else we won't merge your code.

File: chall04/rarias-p.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+						# object which has no lenth, that is the one causing 
+						# the problem here. This only means we are getting a
+						# non numeric value from the file.
+				
+	else:
+		try:
+			lines = []
+			for line in fileinput.input():
+				lines.append(line)
+				pass
```
Comment (aguimbao) [inline chall04/rarias-p.py:84]: Why `pass`?

File: chall04/rarias-p.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+			lenths = checker(lines)
+			printer(argv, "stdin", lenths)
+		except TypeError:
+			printError("stdin", argv)
+
+if __name__ == '__main__':
+	main(sys.argv)
```
Comment (aguimbao) [inline chall04/rarias-p.py:91]: Why pass the `sys.argv` parameter when you can access it from anywhere?

File: chall04/rarias-p.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		possible = []
+		for bookshelf in power(shelves):
+			if listsum(bookshelf) > totalbooks:
+				possible.append(bookshelf)
+		lenths = []
+		for bookshelf in possible:
+			lenths.append(len(bookshelf))
+		return (lenths)
+	except ValueError:
+		pass			# This error is produces another error (line 79) which
```
Comment (aguimbao) [inline chall04/rarias-p.py:55]: Really bad practice to `pass` on an exception.

File: chall04/rarias-p.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+						# come up with.
+
+def main(argv):
+	if len(argv) > 1:
+		fn = 1
+		for file in argv[1:]:
+			try:
+				f = open(argv[fn], "r")
+				lines = f.readlines()
+				lenths = checker(lines)
```
Comment (aguimbao) [inline chall04/rarias-p.py:67]: You mean `lengths`? 😜

File: chall04/rarias-p.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+				
+	else:
+		try:
+			lines = []
+			for line in fileinput.input():
+				lines.append(line)
+				pass
+			lenths = checker(lines)
+			printer(argv, "stdin", lenths)
+		except TypeError:
```
Comment (aguimbao) [inline chall04/rarias-p.py:87]: You should try to catch `EOFError` too for safety incase I `Ctrl-D` your script.

**Review by aguimbao — COMMENTED** [review]
I don't think it's a good idea to calculate all the possible combination of shelves. You could just calculate the "biggest" space with the smallest number of shelves, and if you don't have enough space, keep adding the next biggest shelf.

Anyway, you are deleting other people's files in your PR, you are also submitting your previous challenges, and worst of all, your file didn't have execution permissions (644) :( I will have to close it.

**Pull request closed without merging**